client/src/components/Home.js

```diff
@@ -0,0 +1,27 @@
+import React, { useContext } from 'react';
+import { Container } from 'react-bootstrap';
+import CoinCard from './CoinCard'
+import CoinLabels from './CoinLabels'
+import TopCoin from './TopCoin'
+import { Context } from '../contexts/DataContext'
+
+function Home() {
+  const [{ coinData, currentNames, sortState, setSortState, topCoin} , setState] = useContext(Context)
+
+  let coinsUpper = currentNames
+
+  return (
+    <Container className='homeCont'>
+      <TopCoin number={0} name={topCoin.name} dailyChange={topCoin.percent_change_24h/100}></TopCoin> 
+      <CoinLabels></CoinLabels>
+          {coinData.map((coin, i) => {
+          return <CoinCard key={i} number={i} name={coinsUpper[i]} currentPrice={coin.current_price[coin.current_price.length-1]} rank={coin.rank[coin.rank.length-1]} dailyChange={coin.percent_change_24h/100} weeklyChange={coin.percent_change_7d/100} coinLogo={coin.currencyLogo} ></CoinCard>
```

Comment:
Building the CoinLabels and CoindCards here in Home, with data coming from Context, is pretty awkward.

Normally, "header" and "rows" of a table should be part of the same component, not have to be pieced together on a particular page.  Imagine you want to add another table with the same layout but with data fetched from elsewhere, or based on some criteria other than "top 10 percent change for the day"? (maybe "top 10 absolute change" for the day?)  It'd be fairly tricky to reuse.

If you tried to put 3 charts on the home page right now the structure would looks roughly like:
  -  DataContext 
       - Home
            - CoinLabels
            - CoinCard
            - CoinLabels
            - CoinCard
            - CoinLabels
            - CoinCard
            
If we had a dedicated table component, we could do better:

  -  DataContext 
       - Home
          - Table (makes CoinLabels, CoinCard internally)
          - Table
          - Table

This also surfaces another issue: It's awkward for the CoinLabels component to have the sort-state live in DataContext.  When you have 3 tables, would DataContext now need to know about all 3 tables and handle sorting them separately? Ick, that file will get huge.

With a bit of care, the sorting state for each table could be stored in useState in <Table> and DataContext wouldn't have to know anything about the sort order of the data.